src/htm/encoders/SimHashDocumentEncoder.cpp

```diff
@@ -56,63 +56,153 @@ namespace htm {
       << "Need only one argument of: 'activeBits' or 'sparsity'.";
     NTA_CHECK(parameters.size > 0u)
       << "Missing 'size' argument.";
+    if ((parameters.tokenFrequencyCeiling > 0) && (parameters.tokenFrequencyFloor > 0)) {
+      NTA_CHECK(parameters.tokenFrequencyCeiling > parameters.tokenFrequencyFloor)
+        << "Argument 'tokenFrequencyCeiling' should be greater than argument 'tokenFrequencyFloor'.";
+    }
+    if (!parameters.tokenSimilarity) {
+      NTA_CHECK(parameters.charFrequencyCeiling == 0)
+        << "Argument 'charFrequencyCeiling' requires argument 'tokenSimilarity'.";
+      NTA_CHECK(parameters.charFrequencyFloor == 0)
+        << "Argument 'charFrequencyFloor' requires argument 'tokenSimilarity'.";
+      if ((parameters.charFrequencyCeiling > 0) && (parameters.charFrequencyFloor > 0)) {
+        NTA_CHECK(parameters.charFrequencyCeiling > parameters.charFrequencyFloor)
+          << "Argument 'charFrequencyCeiling' should be greater than argument 'charFrequencyFloor'.";
+      }
+    }
+    if (!parameters.vocabulary.size()) {
+      NTA_CHECK(!parameters.encodeOrphans)
+        << "Argument 'encodeOrphans' requires argument 'vocabulary'.";
+    }
 
     // Make local copy of arguments for processing
     args_ = parameters;
 
     // process: handle 'sparsity' param instead of 'activeBits' param
-    if(args_.sparsity > 0.0f) {
+    if (args_.sparsity > 0.0f) {
       NTA_CHECK((args_.sparsity >= 0.0f) && (args_.sparsity <= 1.0f))
         << "Argument 'sparsity' must be a float in the range 0.0-1.0.";
       NTA_CHECK(args_.size > 0u)
         << "Argument 'sparsity' requires that the 'size' also be given.";
       args_.activeBits = (UInt) round(args_.size * args_.sparsity);
```

Comment:
also. check that the result after multiplication and rounding is sufficient, `> 0`

Reply:
There's a check down at the bottom already for this, I'd be glad to add another for clarity?

Reply:
if it's checked, then that's fine. thanks!